lib/ansible/modules/cloud/azure/azure_rm_appgw.py

```diff
@@ -0,0 +1,792 @@
+            found = True
+            self.log("Response : {0}".format(response))
+            self.log("Application Gateway instance : {0} found".format(response.name))
+        except CloudError as e:
+            self.log('Did not find the Application Gateway instance.')
+        if found is True:
+            return response.as_dict()
+
+        return False
+
+
+def public_ip_id(subscription_id, resource_group_name, name):
+    """Generate the id for a frontend ip configuration"""
+    return '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Network/publicIPAddresses/{2}'.format(
+        subscription_id,
+        resource_group_name,
+        name
+    )
+
+
+def frontend_ip_configuration_id(subscription_id, resource_group_name, appgw_name, name):
+    """Generate the id for a frontend ip configuration"""
+    return '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Network/applicationGateways/{2}/frontendIPConfigurations/{3}'.format(
+        subscription_id,
+        resource_group_name,
+        appgw_name,
+        name
+    )
+
+
+def frontend_port_id(subscription_id, resource_group_name, appgw_name, name):
+    """Generate the id for a frontend port"""
+    return '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Network/applicationGateways/{2}/frontendPorts/{3}'.format(
+        subscription_id,
+        resource_group_name,
+        appgw_name,
+        name
+    )
+
+
+def backend_address_pool_id(subscription_id, resource_group_name, appgw_name, name):
+    """Generate the id for an address pool"""
+    return '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Network/applicationGateways/{2}/backendAddressPools/{3}'.format(
+        subscription_id,
+        resource_group_name,
+        appgw_name,
+        name
+    )
+
+
+def backend_http_settings_id(subscription_id, resource_group_name, appgw_name, name):
+    """Generate the id for a http settings"""
+    return '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Network/applicationGateways/{2}/backendHttpSettingsCollection/{3}'.format(
+        subscription_id,
+        resource_group_name,
+        appgw_name,
+        name
+    )
+
+
+def http_listener_id(subscription_id, resource_group_name, appgw_name, name):
+    """Generate the id for a http listener"""
+    return '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Network/applicationGateways/{2}/httpListeners/{3}'.format(
+        subscription_id,
+        resource_group_name,
+        appgw_name,
+        name
+    )
+
+
+def compare_arrays(old_params, new_params, param_name):
+    old = old_params.get(param_name) or []
+    new = new_params.get(param_name) or []
+
+    oldd = {}
+    for item in old:
+        name = item['name']
+        oldd[name] = item
+    newd = {}
+    for item in new:
+        name = item['name']
+        newd[name] = item
+
+    newd = dict_merge(oldd, newd)
+    return newd == oldd
+
+
+def dict_merge(a, b):
```

Comment:
Was this recently added to another module_util, I seem to recall it being used somewhere else.